src/main/java/org/myrobotlab/kinematics/DHLink.java

```diff
@@ -41,10 +41,10 @@
 
   public transient final static Logger log = LoggerFactory.getLogger(DHLink.class);
   
-  private double velocity;
-  private int state = Servo.SERVO_EVENT_STOPPED;
+  private double velocity; // FIXME - is this set by IK being dp/dt ? .. it should be
```

Comment:
DHLink should not have velocity or state information in it.  I'm not sure where this came from.

Reply:
It his me that add those info needed for collision detection simulation. To be able to do simulation, it need to know the speed (velocity) that this particuliar dhlink can move. I don't think those info affect anything about how IK3D behave. Maybe I should have use other naming to avoid confusion